recipes/CommonVoice/ASR/transformer/train.py

```diff
@@ -268,16 +269,17 @@ def on_fit_start(self):
         # Wrap modules with parallel backend after jit
         self._wrap_distributed()
 
-        # Initialize optimizers after parameters are configured
-        self.init_optimizers()
-
-        # Load latest checkpoint to check to current epoch number
         if self.checkpointer is not None:
             self.checkpointer.recover_if_possible()
 
-        # if the model is resumed from stage two, reinitialize the optimizer
+        # Initialize optimizers after parameters are configured
+        self.init_optimizers()
+
         current_epoch = self.hparams.epoch_counter.current
         if current_epoch > self.hparams.stage_one_epochs:
+            logger.info(
+                f"The checkpoint's epoch(= {current_epoch}) is greater than stage_one_epochs(= {self.hparams.stage_one_epochs}). Using SGD as the optimizer instead of Adam"
```

Comment:
I would rewrite it to something like this so that it is more general:
```
f"The checkpoint's epoch(= {current_epoch}) is greater than stage_one_epochs(= {self.hparams.stage_one_epochs}). Using the fine-tuning optimizer instead of the training one."
```